### PR TITLE
Preserve auditd.log.record_type

### LINKED ISSUE #10830: [Auditd]: auditd.log.record_type not preserved
### Integration Name

Auditd Logs [auditd]

### Dataset Name

auditd.log

### Integration Version

3.20.0

### Agent Version

8.13.0

### Agent Output Type

elasticsearch

### Elasticsearch Version

8.13.0

### OS Version and Architecture

Linux

### Software/API Version

_No response_

### Error Message

`auditd.log.record_type` is missing from `type=SYSCALL` messages

### Event Original

_No response_

### What did you do?

default

### What did you see?

i see `event.action` being set to `syscall`, instead of `auditd.log.record_type` being `SYSCALL`.

### What did you expect to see?

I expected auditd.log.record_type to be there.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by nicholasberlin**
## Proposed commit message
Preserve auditd.log.record_type

Rather than renaming `auditd.log.record_type` to `event.action` and silently failing if `event.action` was pre-existing, set `event.action` with a copy of `auditd.log.record_type` if `event.type` is emtpy. This will preserve `auditd.log.record_type` if `event.type` is empty.

~Also, since the mappings are name instead of number based, fallback to `auditd.log.SYSCALL` to gather syscall ecs mappings if `auditd.log.syscall` fails.~ <-- This needs more work

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] ~~I have verified that all data streams collect metrics or logs.~~
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] ~~I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).~~

## Related issues
- Closes #10830

### CODE REVIEW

File: packages/auditd/data_stream/log/_dev/test/pipeline/test-auditd-enriched.log-expected.json [hunk 1]
```diff
@@ -69,6 +72,92 @@
...
+                    "SGID": "oinstall",
+                    "SUID": "dllogc",
+                    "SYSCALL": "unlink",
+                    "UID": "dllogc",
+                    "a0": "7f1118081d10",
+                    "a1": "7f1118081d10",
+                    "a2": "242",
+                    "a3": "180",
+                    "items": "2",
+                    "key": "delete\"\u001dARCH=x86_64",
```
Comment (andrewkroh) [inline packages/auditd/data_stream/log/_dev/test/pipeline/test-auditd-enriched.log-expected.json:95]: That looks wrong. But it's unrelated to this PR.

**Review by andrewkroh — APPROVED** [review]
LGTM

File: packages/auditd/data_stream/log/_dev/test/pipeline/test-auditd-enriched.log-expected.json (hunk 1 above)
Comment (nicholasberlin) [reply packages/auditd/data_stream/log/_dev/test/pipeline/test-auditd-enriched.log-expected.json:95]: I made an issue here: https://github.com/elastic/integrations/issues/10852
And another, related to parsing the syscall: https://github.com/elastic/integrations/issues/10853

**Pull request merged**